server/dao/order.js

```diff
@@ -53,3 +53,15 @@ exports.getOrderByID = async (db, orderID) => {
 exports.deleteOrder = async (db, orderID) => {
   return db.collection(orderCollectionName).deleteOne(ObjectId(orderID));
 };
+
+// -------------
+// CompleteOrder
+// -------------
+
+exports.completeOrder = async (db, orderID) => {
+  const query = { _id: ObjectId(orderID), status: OrderStatus.PREPARED };
+  const update = { $set: { status: OrderStatus.DONE } };
+  return db
+    .collection(orderCollectionName)
+    .findOneAndUpdate(query, update, { returnDocument: "after" });
```

Comment:
just make the update

```suggestion
    .updateOne(query, update);
```